### DISCUSSION TITLE
How should I move commits up and down multiple times?

### DISCUSSION THREAD

**Discussion by IcyAlmond** [jesseduffield/lazygit#4001] [Q&A] [answered]
I want to move the commit that says "emoji" in the title to just above the highlighted line because that commit is a really small commit which adds an emoji on a line.
<img width="565" alt="{3B0CBBEF-029B-47F6-8D77-440D62AB8F29}" src="https://github.com/user-attachments/assets/1a4e19f7-6d81-43b9-8dda-0570dfe75a3d">
But when I hover over the emoji commit, then `ctrl+j` to move the commit down, I instantly get a merge conflict.

Is there a better way to do what I'm trying to do? If not, will I have to solve a merge conflict for every commit between the highlighted commit and the "emoji" commit?

**Comment by stefanhaller** [comment, marked as answer]
If you want to move a commit down over a longer distance, the best way to do that is to highlight the line just below where you want to move it to (already highlighted in your screenshot), press `e` to start an interactive rebase from there, select the commit(s) you want to move, and press and hold `ctrl-j` to move it down using auto-repeat. It will not move past the "you are here" commit, so you don't have to pay attention to how far you go. Then continue the rebase, and you'll have to resolve the conflict only once.

I see that you have a lot of merge commits between the two commits; I'm not totally sure how well this method works in that case, but I think it should still work fine. I personally find rebasing across merge commits confusing, so I try to avoid that in general. A good way to avoid it is to never merge master into feature branches; always rebase the feature branch onto master instead, this results in a much clearer history that's easier to work with.
